checks/run_checks.go

```diff
@@ -87,3 +92,9 @@ func filterSeverity(level Severity, diagnostics []Diagnostic) []Diagnostic {
 	}
 	return ret
 }
+
+// CheckData is the output returned by the Run function
+type CheckData struct {
+	Diagnostics []Diagnostic
+	Durations   map[string]string
```

Comment:
I'd prefer `map[string]time.Duration` for this, so that API consumers can use the duration values however they like. If that results in gross JSON output we can provide a `MarshalJSON` method for the type to produce a nice value for the durations.